pkg/etcdcli/etcdcli.go

```diff
@@ -208,6 +211,9 @@ func (g *etcdClientGetter) MemberAdd(ctx context.Context, peerURL string) error
 		return err
 	}
 
+	g.clientLock.RLock()
```

Comment:
would that be sufficient? I can still see one scenario where after getting this `cli` here and in between the RLock another goroutine can close the g.client in the meantime. It's less likely than before, but can still happen. 

I think the only way to work around this is to directly depend on the `g.cachedClient` instead of the returned client.

Reply:
Arguably better than what we have now, but yes this is still susceptible to the client being closed out before `Rlock()`.